lib/msf/core/exploit_driver.rb

```diff
@@ -53,9 +53,14 @@ def target_idx
   #
   def compatible_payload?(payload)
     # Try to use the target's platform in preference of the exploit's
-    exp_platform = exploit.targets[target_idx].platform || exploit.platform
+    target = exploit.targets[target_idx]
+    exp_platform = target.platform || exploit.platform
+    exp_arch = target.arch || exploit.arch
 
-    return ((payload.platform & exp_platform).empty? == false)
+    supported_platforms = payload.platform & exp_platform
+    supported_arch = payload.arch & exp_arch
+
+    !supported_platforms.empty? && !supported_arch.empty?
```

Comment:
I'm guessing there's reasons for this not being supported previously? Nothing stood out to me in the git history

Reply:
![](https://media3.giphy.com/media/GcSqyYa2aF8dy/giphy.gif)